rholang/src/main/scala/coop/rchain/rholang/interpreter/SystemProcesses.scala

```diff
@@ -1,284 +1,293 @@
 package coop.rchain.rholang.interpreter
 
+import cats.effect.Async
+import cats.implicits._
 import com.google.protobuf.ByteString
 import coop.rchain.crypto.hash.{Blake2b256, Keccak256, Sha256}
 import coop.rchain.crypto.signatures.{Ed25519, Secp256k1}
 import coop.rchain.models.Expr.ExprInstance.{GBool, GByteArray}
 import coop.rchain.models._
 import coop.rchain.models.rholang.implicits._
-import coop.rchain.rholang.interpreter.Runtime.RhoISpace
+import coop.rchain.rholang.interpreter.Runtime.ShortLeashParams.ShortLeashParameters
+import coop.rchain.rholang.interpreter.Runtime.{BlockTime, RhoISpace, ShortLeashParams}
 import coop.rchain.rholang.interpreter.accounting.Cost
 import coop.rchain.rholang.interpreter.errors.OutOfPhlogistonsError
 import coop.rchain.rholang.interpreter.storage.implicits.matchListPar
 import coop.rchain.rspace.util._
 import coop.rchain.rspace.{ContResult, Result}
-import monix.eval.Task
 
 import scala.util.Try
 
+//TODO: Make each of the system processes into a case class,
+//      so that implementation is not repetitive.
+//TODO: Make polymorphic over match type.
+trait SystemProcesses[F[_]] {
+  def stdOut: (Seq[ListParWithRandomAndPhlos], Int) => F[Unit]
+  def stdOutAck: (Seq[ListParWithRandomAndPhlos], Int) => F[Unit]
+  def stdErr: (Seq[ListParWithRandomAndPhlos], Int) => F[Unit]
+  def stdErrAck: (Seq[ListParWithRandomAndPhlos], Int) => F[Unit]
+  def secp256k1Verify: (Seq[ListParWithRandomAndPhlos], Int) => F[Unit]
+  def ed25519Verify: (Seq[ListParWithRandomAndPhlos], Int) => F[Unit]
+  def sha256Hash: (Seq[ListParWithRandomAndPhlos], Int) => F[Unit]
+  def keccak256Hash: (Seq[ListParWithRandomAndPhlos], Int) => F[Unit]
+  def blake2b256Hash: (Seq[ListParWithRandomAndPhlos], Int) => F[Unit]
+  def getDeployParams(
+      deployParameters: ShortLeashParams[F]
+  ): (Seq[ListParWithRandomAndPhlos], Int) => F[Unit]
+  def blockTime(timestamp: BlockTime[F]): (Seq[ListParWithRandomAndPhlos], Int) => F[Unit]
+}
+
 object SystemProcesses {
 
-  private val prettyPrinter = PrettyPrinter()
+  def apply[F[_]](
+      dispatcher: Dispatch[F, ListParWithRandomAndPhlos, TaggedContinuation],
+      space: RhoISpace[F]
+  )(implicit F: Async[F]): SystemProcesses[F] =
```

Comment:
Isn't `Sync` enough to bring `delay` into scope?

Reply:
You're right. I thought that I would need `Async` later to pass to `Dispatch`, but I don't. I just need it to be a monad for `Parallel[F,G]`.